common/minimodel/__init__.py

```diff
@@ -1002,3 +1002,18 @@ def total_used(self) -> timedelta:
 
     def observations(self) -> List[Observation]:
         return self.root_group.observations()
+
+class Plan:
+    """
+    A 'plan' is a collection of nighly plans
+    """
+    def __init__(self, night_duration=10):
+        self._time_slots_left = night_duration 
+        self._groups = []
+        
+    def add_group(self, group):
+        self.groups.append(group)
+        self._time_slots_left-=1 # TODO: clearly a missrepresantion between time allocates and plan timeslots, a proper function is in place
+    
+    def is_full(self):
+        return self._time_slots_left == 0
```

Comment:
I'm not quite sure how this is intended to work? I've been struggling to think of a name for the Scheduler output since the reinterpretation of nightly plans but plan seems as good a name as anything.

What I'm not clear about here is the significance of `night_duration` (number of time slots in a night?) and why adding a group reduces the number of time slots left by exactly one.

Where is the concept of multiple nights represented here?